src/auxiliary/math.jl

```diff
@@ -5,6 +5,14 @@
 @muladd begin
 #! format: noindent
 
+# `AbstractFloat` clashes with `ForwardDiff.Dual` => use `Real`
+sqrt_(x::T) where {T <: Real} = x < zero(x) ? oftype(x, NaN) : Base.sqrt(x)
```

Comment:
```suggestion
sqrt_(x::Real) = x < zero(x) ? oftype(x, NaN) : Base.sqrt(x)
```
You don't use the type parameter `T` here.